src/policies/conditions.ts

```diff
@@ -443,16 +457,34 @@ export class ConditionContext {
   }
 
   public toJson = async (): Promise<string> => {
-    const userAddressParam = this.conditions
-      .toString()
-      .includes(':userAddress');
-    if (!userAddressParam) {
-      return JSON.stringify({});
+    const payload: Record<string, CustomContextParam | TypedSignature> = {};
+    if (this.conditions.toString().includes(USER_ADDRESS_PARAM)) {
+      payload[USER_ADDRESS_PARAM] = await this.getOrCreateWalletSignature();
     }
-    const typedSignature = await this.getOrCreateWalletSignature();
-    const payload = { ':userAddress': typedSignature };
+
+    const conditions = JSON.parse(this.conditions.toString());
+    conditions.forEach((cond: { parameters: string[] }) => {
+      cond.parameters.forEach((key) => {
+        if (
+          !(key in this.customParameters) &&
+          !SPECIAL_CONTEXT_PARAMS.includes(key)
+        ) {
+          throw new Error(`Missing custom context parameter ${key}`);
+        }
+      });
+    });
+
+    Object.keys(this.customParameters).forEach((key) => {
+      payload[key] = this.customParameters[key];
+    });
     return JSON.stringify(payload);
   };
```

Comment:
@derekpierre Here are the changes that are most relevant to this PR: 
- Seperating variables into  `SPECIAL_CONTEXT_PARAMS` with reserved names
- Adding support for a user-provided map of custom context params